src/annoylib.h

```diff
@@ -164,8 +148,14 @@ struct Angular {
       return random->flip();
   }
   static inline void create_split(const vector<node*>& nodes, int f, Randomness<T>* random, node* n) {
+    // Sample two random points from the set of nodes
+    // Calculate the hyperplane equidistant from them
+    int i = random->index(nodes.size());
+    int j = random->index(nodes.size());
```

Comment:
@erikbern The O(k) method is to sample k points as well as a random hyperplane h. Then, project each of the k points to the random hyperplane (in 2 dimensions this is just a bunch of points on a line). It should now be trivial to find a second hyperplane h' perpendicular to h that splits the projected points and will in turn split the points in the original space. 

I tried to draw a picture of this. Not sure if it came through.

![image](https://cloud.githubusercontent.com/assets/928322/7815632/9f13c844-0397-11e5-8901-6e809ebd7a07.png)